src/handler/mod.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::{Error, Result};
+use figlet_rs::FIGfont;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::process::Command;
+use std::str;
+
+pub fn handle_reboot(force: bool) -> Result<(), Error> {
+    let mut boot_counter: &str = "";
+    if !force {
+        let grub_vars = Command::new("grub2-editenv")
+            .arg("-")
+            .arg("list")
+            .output()?;
+        let grub_vars = str::from_utf8(&grub_vars.stdout[..]).unwrap().split('\n');
+
+        for var in grub_vars {
+            if var.contains("boot_counter") {
+                boot_counter = var.split('=').last().unwrap();
+            }
+        }
+
+        if boot_counter.parse::<i64>().unwrap() <= -1 {
+            log::error!("cannot reboot as boot_counter is -1 ");
```

Comment:
if for whatever reason `boot_counter` is not modified by the loop above the `unwrap` will fail and we will not get to the `Ok`

Reply:
changed it with "?"